lib/docker.ts

```diff
@@ -132,69 +180,101 @@ class Docker {
   }
 
   /**
-   * Analyze and return key files textual content and MD5 sum.
-   * @param imageTarPath path to image file saved in tar format
-   * @param txtPatterns list of plain text key files paths patterns to extract and return as strings
-   * @param md5Patterns list of binary key files paths patterns to extract and return their MD5 sum
-   * @returns key files textual content and MD5 sum
+   * Convenience function to wrap save to tar and extract files from tar
+   * @param extractActions array of pattern, callbacks pairs
+   * @returns extracted file products
    */
-  public async analyze(
-    imageTarPath: string,
-    txtPatterns: string[],
-    md5Patterns: string[] = [],
-  ) {
-    const result: ExtractedKeyFiles = { txt: {}, md5: {} };
+  public async extract(
+    extractActions: ExtractAction[],
+  ): Promise<ExtractProductsByFilename> {
+    return this.save(async (err, imageTarPath) => {
```

Comment:
change to async await

Reply:
maybe I miss something, I believe it's already in async-await form ?!?